ansible_base/authentication/utils/claims.py

```diff
@@ -905,11 +906,17 @@ def cache_existing(self, role_assignments):
             # Cache Role User Assignment
             self._init_cache_key(role_definition.name, content_type_id=role_assignment.content_type_id)
 
-            # object_id is TEXT db type
-            object_id = int(role_assignment.object_id) if role_assignment.object_id is not None else None
-            obj = role_assignment.content_object if object_id else None
-
-            self.cache[role_definition.name][role_assignment.content_type_id][object_id] = {'object': obj, 'status': self.STATUS_EXISTING}
+            # Only should cache local role assignments
+            if role_assignment.content_type is None or role_assignment.content_type.service in [get_local_resource_prefix(), "shared"]:
```

Comment:
If the `role_assignment.content_type is None` case is hit, then it's expected that object_id will be blank, which will lead to erroneous `logger.exception` calls. It looks like the caller method `manage_permissions` handles those separately, so I would say change this

```suggestion
            if (role_assignment.content_type is not None) and role_assignment.content_type.service in [get_local_resource_prefix(), "shared"]:
```

to avoid log spam